docker/README.md

```diff
@@ -0,0 +1,104 @@
+## Running with Docker
+
+To simplify running GraphQL Mesh, you can use the Docker image and the docker compose template we
+provide. This setup allows you to easily configure and run the gateway with your own environment
+variables without the need to install node and the required mesh npm packages.
+
+### Prerequisites
+
+Ensure you have Docker and Docker Compose installed on your machine. If not, you can download and
+install them from the official Docker website.
+
+### Docker Compose Setup
+
+1. **Create a `docker-compose.yml` file with the following:**
+
+   ```yaml
+   version: '3.8'
+
+   services:
+     graphql-mesh:
+       image: theguild/graphql-mesh:v1
+       ports:
+         - '4000:4000'
+       environment:
+         HIVE_CDN_ENDPOINT: SECRET
+         HIVE_CDN_KEY: SECRET
+         HIVE_REGISTRY_TOKEN: SECRET
+       security_opt:
+         - no-new-privileges:true
+       cap_drop:
+         - ALL
+       cap_add:
+         - CHOWN
+       read_only: true
+       volumes:
+         - /tmp # make /tmp writable
+         # Uncomment the line below to inject your custom mesh.config.ts
+         # - ./path/to/local/mesh.config.ts:/app/mesh.config.ts:ro
+   ```
+
+2. **Replace placeholders:**
+
+- Replace #### in `HIVE_CDN_ENDPOINT`, `HIVE_CDN_KEY`, and `HIVE_REGISTRY_TOKEN` with your actual
+  values.
+
+3. **Run Docker Compose:**
+
+- In the directory containing your `docker-compose.yml` file, run the following command:
+
+  ```sh
+  docker compose up
+  ```
+
+  This command will:
+
+  - Pull the graphql mesh gateway image if it is not already available locally.
+  - Pull the supergraph from your Hive registry using the provided environment variables.
+  - Start the Mesh Gateway service on port `4000`.
+
+### Custom Docker Image
+
+You can create a custom Docker image that inherits from the theguild/graphql-mesh:v1 base image.
+
+Example Dockerfile:
+
+```dockerfile
+FROM theguild/graphql-mesh:v1
+
+# Set a non-root user to run the container
+RUN useradd --create-home --shell /bin/bash mesh
```

Comment:
I think all of these lines are redundant when you extend?

Reply:
you're right, only the first line is needed, fixed